src/GitTfs/Commands/ExportMap.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.ComponentModel;
+using NDesk.Options;
+using Newtonsoft.Json;
+using StructureMap;
+
+namespace GitTfs.Commands
+{
+    [Pluggable("exportmap")]
+    [Description("exportmap -f <file> -r <repo>")]
+    public class ExportMap : GitTfsCommand
+    {
+        private readonly Globals _globals;
+        private readonly Init _init;
+        private readonly Help _helper;
+
+        public ExportMap(Globals globals, Init init, Help helper)
+        {
+            _globals = globals;
+            _init = init;
+            _helper = helper;
+        }
+
+        public string FilePath { get; set; }
+        public string RepositoryPath { get; set; }
+
+        public OptionSet OptionSet
+        {
+            get
+            {
+                return new OptionSet
+                {
+                     { "f|file=", "The output file path",
+                        f => FilePath = f },
+                     { "r|repository=", "The path to the repository",
+                        r => RepositoryPath = r }
+                };
+            }
+        }
+
+        public int Run()
+        {
+            if (string.IsNullOrWhiteSpace(RepositoryPath)
+                || string.IsNullOrWhiteSpace(FilePath))
+            {
+                return _helper.Run(this);
+            }
+
+            _globals.Repository = _init.GitHelper.MakeRepository(RepositoryPath);
+            var commits = _globals.Repository.GetCommitChangeSetPairs();
+            var dic = JsonConvert.SerializeObject(commits);
```

Comment:
Is it really necessary to do this with JSON ?
Wouldn't a CSV, or just a file with changeset=commit-hash lines suffice ?

If there are genuine reasons to use JSON, then pulling in a new dependency might be justified, but if any other (simpler) format would suffice maybe we should go for that ?

Reply:
it is not necessary to use JSON, i was used to it and therefore i included it.